v0.3/tutorials/quickstart 1 button stopwatch.eve

```diff
@@ -0,0 +1,198 @@
+    [#svg/circle #second-face sort: 1, cx: 50, cy: 50, r: 48, fill: "#fff", stroke: "#000", stroke-width: 0.5]
+    [#svg/circle #minute-face sort: 2, cx: 50, cy: 26, r: 12, fill: "#fff", stroke: "#000", stroke-width: 0.25]
+    [#minute-hand sort: 3, degrees: 6 * minute, length: 11, stroke: "#000", stroke-width: 0.5]
+    [#second-hand sort: 4, degrees: 6 * second + 0.006 * millisecond, length: 46, stroke: "#ff0000", stroke-width: 0.5]]
+```
+
+This block closely resembles its predecessor, but the search looks for `#race-timer` now instead of the system time since it's the race time that will determine the position of the hands.
+
+## One Button
+
+### Draw the button
+
+All we need to do is add one button to cycle between start, stop, and reset.
+
+```
+search
+  [#app page: "Clock"]
+  view = [#qs-contents]
+
+bind
+  view.children += [#ui/div sort: 2 children:
+    [#ui/div #qs-stopwatch-btn text: "Start/Stop/Reset"]]
+```
+
+### Pressing Start
+
+```
+search
+  [#time timestamp]
+  race-timer = [#race-timer #stopped #reset]
+  [#html/event/click element: [#qs-stopwatch-btn]]
+
+commit
+  [#start-time timestamp]
+  race-timer -= #stopped
+  race-timer -= #reset
+  race-timer += #running
+```
+
+If the button is pressed while the stopwatch is stopped and reset (thus at zero), this records the `#start-time` and changes the `#race-timer` from `#stopped` to `#running` and removes the `#reset` tag.
+
+### Pressing Stop
+
+```
+search
+  [#time timestamp]
+  race-timer = [#race-timer #running]
+  [#html/event/click element: [#qs-stopwatch-btn]]
+
+commit
+  [#stop-time timestamp]
+  race-timer += #stopped
+  race-timer -= #running
+```
+
+If the button is pressed with the stopwatch is running, this records the `#stop-time` and changes the `#race-timer` from `#running` to `#stopped`.
+
+### Pressing Reset
+
+```
+search
+  race = [#race-timer #stopped not(#reset)]
+  start = [#start-time]
+  stop = [#stop-time]
+  [#html/event/click element: [#qs-quickstart-btn]]
+
+commit
+  race := none
+  start := none
+  stop := none
+  [#race-timer #stopped #reset]
+```
+
+When the timer is stopped but hasn't been reset yet, and the button gets hit, this erases the `#race-timer`, `#start-time`, and `#stop-time` records to clear the stopwatch of the previous time's data, and recommits a new `#race-timer` without any time values on it.
+
+## Adding time values to the race timer
+
+There's three conditions where the `#race-timer` needs values assigned to it.
+
+### Time values while reset
+
+```
+search
+  race-timer = [#race-timer #stopped #reset]
+
+bind
+  race-timer.minute += 0
+  race-timer.second += 0
+  race-timer.millisecond += 0
+```
+
+Before the stopwatch button is pressed, there are no time values anywhere, so we just manually set three attributes to the `#race-timer` for minutes, seconds, and milliseconds, and make those all 0. This way, when the page is loaded or the stopwatch is reset, the hands are zeroed out.
```

Comment:
This code isn't actually setting a default, you should end up with two values whenever you reset, unless you're handling the removal of the other values elsewhere, That may incidentally work since the SVG guy will overwrite attributes last write wins, but it's probably not a style we should encourage.